tests/validation/tests/v3_api/test_monitoring.py

```diff
@@ -0,0 +1,299 @@
+    validate_cluster_graph(query1, "kube-component")
+
+
+# rancher component graphs are from the fluent app for cluster logging
+def test_monitoring_rancher_component_graph():
+    rancher_client, cluster = get_admin_client_and_cluster()
+    # check if the cluster logging is enabled, assuming fluent is used
+    if cluster.enableClusterAlerting is False:
+        print("cluster logging is not enabled, skip the test")
+        return
+    else:
+        cluster_monitor_obj = rancher_client.list_clusterMonitorGraph()
+        # generate the request payload
+        query1 = copy.deepcopy(cluster_query_template)
+        query1["obj"] = cluster_monitor_obj
+        query1["filters"]["clusterId"] = cluster.id
+        query1["filters"]["displayResourceType"] = "rancher-component"
+        validate_cluster_graph(query1, "rancher-component")
+
+
+def test_monitoring_node_graph():
+    rancher_client, cluster = get_admin_client_and_cluster()
+    node_list_raw = rancher_client.list_node(clusterId=cluster.id).data
+    for node in node_list_raw:
+        cluster_monitor_obj = rancher_client.list_clusterMonitorGraph()
+        # generate the request payload
+        query1 = copy.deepcopy(cluster_query_template)
+        query1["obj"] = cluster_monitor_obj
+        query1["filters"]["clusterId"] = cluster.id
+        query1["filters"]["resourceType"] = "node"
+        query1["metricParams"]["instance"] = node.id
+        validate_cluster_graph(query1, "node")
+
+
+def test_monitoring_workload_graph():
+    rancher_client, cluster = get_admin_client_and_cluster()
+    system_project = rancher_client.list_project(clusterId=cluster.id, name="System").data[0]
+    project_monitor_obj = rancher_client.list_projectMonitorGraph()
+    # generate the request payload
+    query1 = copy.deepcopy(cluster_query_template)
+    query1["obj"] = project_monitor_obj
+    query1["filters"]["projectId"] = system_project.id
+    query1["filters"]["resourceType"] = "workload"
+    query1["metricParams"]["workloadName"] = \
+        "deployment:cattle-prometheus:grafana-cluster-monitoring"
+    validate_cluster_graph(query1, "workload")
+
+
+def test_monitoring_enable_project_monitoring():
+    rancher_client, cluster = get_admin_client_and_cluster()
+    system_project = rancher_client.list_project(name="System").data[0]
+    system_project_client = get_project_client_for_token(system_project, ADMIN_TOKEN)
+    project = namespace["project"]
+    # enable the project monitoring
+    project.enableMonitoring()
+    project_client = get_project_client_for_token(project, ADMIN_TOKEN)
+    namespace["project"] = rancher_client.reload(project)
+    wait_for_app_to_active(project_client, PROJECT_MONITORING_APP)
+    wait_for_app_to_active(system_project_client, MONITORING_OPERATOR_APP)
+    namespace["project"] = rancher_client.reload(project)
+    # wait for targets to be up
+    wait_for_target_up("expose-prometheus-metrics")
+    wait_for_target_up("expose-grafana-metrics")
```

Comment:
@jiaqiluo Can we move this test as part of test_monitoring_workload_metrics tests ?

Reply:
fixed.